lib/route53.js

```diff
@@ -7,6 +7,52 @@ const Promise = require("bluebird");
 const CustomError = require("./errors")
 const DNSHandler = require("./base")
 
+
+function changeRequestMethod(params, method) {
+	params.ChangeBatch.Changes[0].Action = method;
+	return params;
+}
+
+function generateParams(hostedZoneId, domain, opts) {
+	let params = {
+		HostedZoneId: hostedZoneId,
+		ChangeBatch: {
+			Changes: [
+				{
+					Action: "",
+					ResourceRecordSet: {
+						Name: opts.name.concat(".", domain, "."),
+						Type: opts.type || "CNAME",
+						TTL: opts.ttl || 3600,
+					},
+				},
+			],
+		},
+	};
+
+	if (opts.values) {
+		params.ChangeBatch.Changes[0].ResourceRecordSet.ResourceRecords = opts.values.map(function(r){
+			return {
+				Value: r
+			}
+		});
+	}
+	return params;
+}
+
+function checkForDomainInResponse(hostName, response, domain) {
+	let record = _.filter(_.flatMap(response), {
```

Comment:
checkout [_.some()](https://lodash.com/docs/4.17.4#some) for checking an array for any matches. Should be more performant since it will quit as soon as it finds one match.